### PR TITLE
Option for local plus codes on osd

### LINKED ISSUE #5114: Option to chop off the first four digits of the plus code, since they're optional
## Current Behavior
Currently, INAV shows more than 10 digits of a plus code. The first four digits are optional and can be omitted by specifying a general location (like the city you're flying near).

## Desired Behavior

Adding an option to not show the first four digits would make the plus code display much shorter and more convenient.

## Who does this impact? Who is this for?
Anyone who uses plus codes could benefit from this.

### PR DESCRIPTION
**Pull request by avsaase**
This PR resolves #5114 by implementing a CLI command `osd_plus_code_type` with options `GLOBAL` (the default) and `LOCAL`, where `LOCAL` chops off the first four digits of the plus code. To find the original location you'll need a reference point like a nearby city or mountain within 0.4 degrees longitude and 0.4 latitude, roughly 40 km.

~~The configurator does not yet understand this option so it doesn't allow placing the shorter plus code against the right edge of the screen. To position the osd element like in this screengrab you need to use CLI command `osd_layout 0 97 21 12 V`.~~ Configurator side: iNavFlight/inav-configurator#1150.

2021-01-03 11_41_35-PICT0010 AVI - VLC Media Player

### CODE REVIEW

**Comment by avsaase** [comment]
I'm struggling updating the docs. Running `python src\utils\update_cli_docs.py` gives me
```
Traceback (most recent call last):
  File "src\utils\update_cli_docs.py", line 4, in <module>
    import yaml  # pyyaml / python-yaml
ModuleNotFoundError: No module named 'yaml'
```
I have `PyYAML` installed. Am I doing something wrong?

**Comment by skorokithakis** [comment]
This is fantastic, thanks for implementing it!

**Comment by avsaase** [comment]
Do you think it could be useful to also have options to chop of 2 or 6 digits? Removing two digits would work for basically everyone in the RC hobby and 6 digits could be useful if you only fly around a field.

**Comment by skorokithakis** [comment]
Personally I think removing 6 digits is overkill, and I always like to have fewer knobs with sane defaults, rather than more. I think 4 digits plus a feature within 40km is fine, and I *think* Google maps has better support for that, though I'm not entirely sure about that.

**Comment by avsaase** [comment]
I've checked and google maps works for any shortened plus code. 2km to the reference point is definitely too far for long range flyers but for those that stay within VLOS (all of us of course) this allows to save a bit more OSD space.

**Comment by skorokithakis** [comment]
If it would be more useful to you, that's fine with me (but I'm also fine with the longer code). Then again, I'm not an INAV developer so I have no authority.

**Comment by avsaase** [comment]
Using the text-based setting `osd_plus_code_type = GLOBAL/COUNTRY/LOCAL/FIELD` was a poor design choice in retrospect, so I changed it to a direct selection of the number of leading digits removed: `osd_plus_code_short = 0/2/4/6`. Updated the configurator side accordingly.

File: src/main/fc/settings.yaml [hunk 1]
```diff
@@ -155,6 +155,9 @@ tables:
   - name: nav_overrides_motor_stop
     enum: navOverridesMotorStop_e
     values: ["OFF_ALWAYS", "OFF", "AUTO_ONLY", "ALL_NAV"]
+  - name: osd_plus_code_short
+    # enum: osd_plus_code_short_e
```
Comment (avsaase) [inline src/main/fc/settings.yaml]: I forgot to remove this line. Should I push a new commit? Is there a cost to running all the checks for every new push?

Comment (DzikuVx) [reply src/main/fc/settings.yaml]: @avsaase please push new commit

**Review by DzikuVx — CHANGES_REQUESTED** [review]
You still have to bump PG version in `PG_REGISTER_WITH_RESET_TEMPLATE(osdConfig_t, osdConfig, PG_OSD_CONFIG, 13);`

**Review by DzikuVx — APPROVED** [review]
@avsaase please resolve conflict and we can merge it right after

**Comment by DzikuVx** [comment]
ah no, no conflict after all, my bad, merging

**Comment by avsaase** [comment]
Thanks Paweł.

**Pull request merged**